plugins/flytekit-ray/flytekitplugins/ray/task.py

```diff
@@ -50,11 +57,31 @@ class RayFunctionTask(PythonFunctionTask):
     _RAY_TASK_TYPE = "ray"
 
     def __init__(self, task_config: RayJobConfig, task_function: Callable, **kwargs):
-        super().__init__(task_config=task_config, task_type=self._RAY_TASK_TYPE, task_function=task_function, **kwargs)
+        super().__init__(
+            task_config=task_config,
+            task_type=self._RAY_TASK_TYPE,
+            task_function=task_function,
+            **kwargs,
+        )
         self._task_config = task_config
 
     def pre_execute(self, user_params: ExecutionParameters) -> ExecutionParameters:
-        ray.init(address=self._task_config.address)
+        init_params = {"address": self._task_config.address}
+
+        ctx = FlyteContextManager.current_context()
+        if not ctx.execution_state.is_local_execution():
+            working_dir = os.getcwd()
+            init_params["runtime_env"] = {"working_dir": working_dir}
+
+            cfg = self._task_config
+            if cfg.excludes_working_dir:
+                init_params["runtime_env"]["excludes"] = cfg.excludes_working_dir
+
+            # fast register data with timestamp mtime=0 will be zipped and uploaded to ray gcs
+            # zip does not support timestamps before 1980 -> hacky workaround of touching all the files
+            os.system(f"touch `find {working_dir} -type f`")
```

Comment:
This depends on the system having `touch` + `find`, which is usually the case.

To be safer, can this use [os.utime](https://docs.python.org/3/library/os.html#os.utime) + [os.walk](https://docs.python.org/3.9/library/os.html#os.walk) to update the mtimes?

Reply:
Good catch! I can try this out & adjust. Also this should only stay short term. @pingsutw mentioned there is work in progress that allows us to not set `mtime=0` for the fast register `tar.gz` cc @pingsutw

Reply:
i do not think we will do this right now @fiedlerNr9, because the mtime=0 makes it possible to get consistent hashes for the similar tar files, otherwise we get multiple uploads to admin.

I think you should do what @thomasjpfan recommends use `os.walk`. the os.system will break and will cause random bugs.

Reply:
No need to manually adjust the timestamp for this now. I just catch the fast register data and dont zip it at all